src/rust/lib/graph-editor/src/component/visualization/container.rs

```diff
@@ -108,12 +216,52 @@ impl ContainerData {
         self.set_visibility(true);
     }
 
-    /// Set the visualization shown in this container..
-    fn set_visualization(&self, visualization:Visualization) {
-        self.add_child(&visualization);
+    /// Set the visualization shown in this container.
+    pub fn set_visualization(&self, visualization:Visualization) {
+        let vis_parent = &self.display_object_visualisation;
+        visualization.display_object().set_parent(&vis_parent);
+
         self.visualization.replace(Some(visualization));
         self.init_visualization_properties();
     }
+
+    fn update_shape_sizes(&self) {
+        let overlay_shape = &self.shape_overlay.shape;
+        let frame_shape   = &self.shape_frame.shape;
+        let padding       = self.padding.get();
+        let width         = self.size.get().x;
+        let height        = self.size.get().y;
+        frame_shape.width.set(width);
+        frame_shape.height.set(height);
+        frame_shape.padding.set(padding);
+        frame_shape.sprite.size().set(Vector2::new(width, height));
+        overlay_shape.width.set(width);
+        overlay_shape.height.set(height);
+        overlay_shape.padding.set(padding);
+        overlay_shape.sprite.size().set(Vector2::new(width, height));
+
+
+        let pos = Vector3::new(width/2.0,height/2.0, 0.0);
+        frame_shape.set_position(pos);
+        overlay_shape.set_position(pos);
+    }
+
+    fn set_size(&self, size:Vector3<f32>) {
+        if let Some(vis) = self.visualization.borrow().as_ref() {
+            vis.set_size(size.xy());
+        }
+
+        self.size.set(size.xy());
+        self.update_shape_sizes();
+    }
+
+    fn set_corner_roundness(&self, value:f32) {
```

Comment:
rename to set_corner_radiuses as this is what we use in other places, like in shapes definitions. We can alternatively rename it there, but I believe that using "radius" is more common (CSS like things)